custom_components/magic_areas/binary_sensor.py

```diff
@@ -253,24 +237,83 @@ def device_class(self):
         """Return the class of this binary_sensor."""
         return DEVICE_CLASS_OCCUPANCY
 
-    def autolight_disable_state_change(self, entity_id, from_state, to_state):
+    async def async_added_to_hass(self):
+        """Call when entity about to be added to hass."""
+        if self.hass.is_running:
+            await self._setup_listeners()
+        else:
+            self.hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_STARTED, self._setup_listeners
+            )
+
+        last_state = await self.async_get_last_state()
+        is_new_entry = last_state is None  # newly added to HA
 
-        if self._passive:
+        if is_new_entry:
+            _LOGGER.debug(f"New area detected: {self.area.slug}")
+            self._update_state()
+        else:
+            _LOGGER.debug(f"Area restored: {self.area.slug} [{last_state.state}]")
+            self._state = last_state.state == STATE_ON
+            self.schedule_update_ha_state()
+
+    async def async_will_remove_from_hass(self):
+        """Remove the listeners upon removing the component."""
+        self._remove_listeners()
+
+    async def _setup_listeners(self, _=None) -> None:
+        _LOGGER.debug("%s: Called '_setup_listeners'", self._name)
+        if not self.hass.is_running:
+            _LOGGER.debug("%s: Cancelled '_setup_listeners'", self._name)
             return
 
+        # Track presence sensors
+        remove_presence = async_track_state_change(
+            self.hass, self.presence_sensors, self.sensor_state_change
+        )
+
+        # Track autolight_disable sensor if available
+        autolights_config = self.area.config.get(CONF_AUTO_LIGHTS)
+        if autolights_config.get(CONF_AL_DISABLE_ENTITY):
+            remove_disable = async_track_state_change(
+                self.hass,
+                autolights_config.get(CONF_AL_DISABLE_ENTITY),
+                self.autolight_disable_state_change,
+            )
+            self.tracking_listeners.append(remove_disable)
+
+        # Timed self update
+        delta = timedelta(seconds=self.area.config.get(CONF_UPDATE_INTERVAL))
+        remove_interval = async_track_time_interval(self.hass, self.update_area, delta)
+
+        self.tracking_listeners.extend([remove_presence, remove_interval])
+
+    def _remove_listeners(self):
+        while self.tracking_listeners:
+            remove_listener = self.tracking_listeners.pop()
+            remove_listener()
+
+    def autolight_disable_state_change(self, entity_id, from_state, to_state):
+
         if to_state.state == STATE_OFF:

```

Comment:
doesn't respect the custom `disable_state`

Reply:
Good catch! Will push a fix soon

Reply:
Actually this bug is on the master so should be a `hotfix/`. Will merge this one and push a hotfix to the master.